src/frequenz/client/base/channel.py

```diff
@@ -85,17 +74,12 @@ class ChannelOptions:
     keep_alive: KeepAliveOptions = KeepAliveOptions()
     """HTTP2 keep-alive options for the channel."""
 
-    sign: SigningOptions | None = None
-    """Signing options for the channel."""
-
-    auth: AuthenticationOptions | None = None
-    """Authentication options for the channel."""
-
 
 def parse_grpc_uri(
     uri: str,
     /,
     defaults: ChannelOptions = ChannelOptions(),
+    interceptors: list[ClientInterceptor] | None = None,
```

Comment:
Nitpick: I would move `interceptors` before the `/,` to allow to use it as a positional argument. We require the keyword for `defaults` because `ChannelOptions` are sometimes used as concrete options and sometimes as defaults that will be overridden by those concrete options, so we wanted to make it extra clear these are just defaults and might not be the options that are used by the channel at the end. But with interceptors there is no ambiguity, instances passed in there will be instances of `ClientInterceptors` and there is only one way to use them and there will be no overrides.

Super nitpick because at the end probably almost nobody will use `parse_grpc_uri` directly, so it won't make much of a difference.